Eatery/Controllers/Eateries/Campus/ScrollableViewController.swift

```diff
@@ -0,0 +1,374 @@
+        filterLabel.text = ""
+        filterLabel.preservesSuperviewLayoutMargins = true
+        headerContainer.addSubview(filterLabel)
+
+        separatorView = SeparatorView(insets: UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 0))
+        separatorView.tintColor = .clear
+        separatorView.isHidden = true
+        headerContainer.addSubview(separatorView)
+
+        titleLabel.snp.makeConstraints { make in
+            make.leading.equalToSuperview().offset(tabBarPadding)
+            make.centerY.equalToSuperview()
+            make.bottom.equalToSuperview()
+        }
+
+        filterButton.snp.makeConstraints { make in
+            make.centerY.equalTo(titleLabel.snp.centerY)
+            make.height.equalTo(titleLabel.snp.height).multipliedBy(0.3)
+            make.trailing.equalToSuperview().offset(-tabBarPadding)
+            make.width.equalTo(filterButton.snp.height).multipliedBy(1.71)
+        }
+
+        filterLabel.snp.makeConstraints { make in
+            make.centerY.equalTo(titleLabel.snp.centerY)
+            make.trailing.equalTo(filterButton.snp.leading).offset(-6)
+        }
+
+        separatorView.snp.makeConstraints { make in
+            make.leading.trailing.bottom.equalToSuperview()
+        }
+
+        let titles = viewControllers.map {
+            $0.category
+        }
+        tabBar = ScrollableTextTabBarControl(sections: titles, padding: 16)
+        tabBar.directionalLayoutMargins = NSDirectionalEdgeInsets(top: 0, leading: 16, bottom: 0, trailing: 16)
+        tabBar.addTarget(self, action: #selector(manualScroll), for: .valueChanged)
+        tabBar.backgroundColor = .white
+        tabBar.layer.zPosition = 2
+
+        tabBar.snp.makeConstraints { make in
+            make.height.equalTo(40)
+        }
+
+        tabBarStack = UIStackView(arrangedSubviews: [headerContainer, tabBar])
+        tabBarStack.axis = .vertical
+    }
+
+    private func createMenuStack() {
+        menuItemsView = UIView()
+        menuItemsView.backgroundColor = .white
+
+        menuItemsStack = UIStackView(arrangedSubviews: originalViews)
+        menuItemsStack.axis = .vertical
+        menuItemsView.addSubview(menuItemsStack)
+        view.addSubview(menuItemsView)
+        view.addSubview(tabBarStack) // Tabbar added here to account for z position
+
+        tabBarStack.snp.makeConstraints { make in
+            make.top.equalToSuperview()
+            make.leading.trailing.equalToSuperview()
+        }
+
+        menuItemsView.snp.makeConstraints { make in
+            make.top.equalTo(tabBar.snp.bottom)
+            make.leading.trailing.bottom.equalToSuperview()
+        }
+
+        menuItemsStack.snp.makeConstraints { make in
+            make.top.bottom.equalToSuperview()
+            make.leading.trailing.equalToSuperview().inset(5)
+        }
+
+    }
+
+    private func createHighToLowStack() {
+        let reversedItems = Array(allItems.reversed())
```

Comment:
I think this array is being reversed because it's high to low? Either way could you add a comment explaining why the items are reversed?